drivers/SmartThings/harman-luxury/src/disco_helper.lua

```diff
@@ -0,0 +1,185 @@
+            for _, raw_text_array in pairs(found_item.txt.text or {}) do
+                local text_item = byte_array_to_plain_text(raw_text_array)
+                table.insert(text_list, text_item)
+            end
+        end
+    end
+
+    local answer_text = find_text_in_answers_by_ip(ip, discovery_responses)
+    for _, text_item in pairs(answer_text or {}) do
+        table.insert(text_list, text_item)
+    end
+    return text_list
+end
+
+local function get_parameters(driver, ip, discovery_responses)
+    local params = {
+        dni = "",
+        mnid = "",
+        setupid = ""
+    }
+    local text_list = find_text_list_in_mdns_response(driver, ip, discovery_responses)
+    for _, text in ipairs(text_list) do
+        for key, value in string.gmatch(text, "(%S+)=(%S+)") do
+            if key == "mac" then
+                local dni = value:gsub("-", ""):gsub(":", ""):lower()
+                log.info("get_parameters : use mac as dni. mac = " .. value .. " | dni = " .. dni)
+                params["dni"] = dni
+            end
+            if key == "mnid" then
+                log.info("get_parameters : mnid = " .. value)
+                params["mnid"] = value
+            end
+            if key == "setupid" then
+                log.info("get_parameters : setupid = " .. value)
+                params["setupid"] = value
+            end
+        end
+    end
+
+    if params["dni"] == nil then
+        log.error("get_parameters : failed to find dni")
+        return nil
+    else
+        return params
+    end
+end
+
+local function insert_dni_ip_from_answers(driver, filtered_responses, target_table)
+    for _, answer in pairs(filtered_responses.answers) do
+        local ip
+
+        if answer.kind.ARecord ~= nil then
+            ip = answer.kind.ARecord.ipv4
+        end
+
+        if ip ~= nil then
+            local params = get_parameters(driver, ip, filtered_responses)
+
+            if params ~= nil then
+                log.info("answer_name, arecod = " .. tostring(answer.name) .. ", " .. tostring(answer.kind.ARecord))
+                local dni = params["dni"]
+                local mnid = params["mnid"]
+                local setupid = params["setupid"]
+                target_table[dni] = {
+                    ip = ip,
+                    mnid = mnid,
+                    setupid = setupid
+                }
+            end
+        end
+    end
+end
+
+local function filter_response_by_servie_name(service_type, discovery_responses)
+    local filtered_responses = {
+        answers = {},
+        found = {}
+    }
+
+    for _, answer in pairs(discovery_responses.answers or {}) do
+        table.insert(filtered_responses.answers, answer)
+    end
+
+    for _, additional in pairs(discovery_responses.additional or {}) do
+        table.insert(filtered_responses.additional, additional)
+    end
+
+    for _, found in pairs(discovery_responses.found or {}) do
```

Comment:
Just making a note: This is the correct structure for the mDNS API; released versions of the API will always use the `.found` field.

The rest of the code should treat the `{ answers, additional }` form as the fallback, not the `found` form.